python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+      - \ref ap_association_AssociationDBSqliteTask_Config
+      - \ref ap_association_AssociationDBSqliteTask_Run
+      - \ref ap_association_AssociationDBSqliteTask_Debug
+      - \ref ap_association_AssociationDBSqliteTask_Example
+
+    \section ap_association_AssociationDBSqliteTask_Purpose   Description
+    Create a simple sqlite database and implement wrappers to store and
+    retrieve DIAObjects and DIASources within that database. This task
+    functions as a testing ground for the L1 database and should mimic this
+    database's eventual functionaly. This specific database implementation is
+    useful for the verification packages which may not be run with access to
+    L1 database.
+
+    \section ap_association_AssociationDBSqliteTaskInitialize
+        Task initialization
+    \copydoc \_\_init\_\_
+
+    \section ap_association_AssociationDBSqliteTask_Run
+        Invoking the Task
+    \copydoc run
+
+    \section ap_association_AssociationDBSqliteTask_Config
+        Configuration parameters
+    See \ref AssociationDBSqliteConfig
+
+    \section ap_association_AssociationDBSqliteTask_Debug
+        Debug variables
+    This task has no debug variables
+
+    \section ap_association_AssociationDBSqliteTask_Example
+        Example of using AssociationDBSqliteTask
+    This task has no standalone example, however it is applied as a subtask of
+    ap.association.AssociationTask
+    """
+
+    ConfigClass = AssociationDBSqliteConfig
+    _DefaultName = "association_db_sqlite"
+
+    def __init__(self, **kwargs):
+        """ Create a new connection to the sqlite database specified in
+        the config.
+        """
+        pipeBase.Task.__init__(self, **kwargs)
+        self.indexer = IndexerRegistry[self.config.indexer.name](
+            self.config.indexer.active)
+        self._db_connection = sqlite3.connect(self.config.db_name)
+        self._db_cursor = self._db_connection.cursor()
+
+    def run(self):
+        """ Not implemented in this subtask. Please use the store and load
+        methods.
+        """
+        Raise(NotImplementedError)
+
+    def commit(self):
+        """ Save changes to the sqlite database.
+        """
+        self._db_connection.commit()
+
+    def close(self):
+        """ Close the connection to the sqlite database.
+        """
+        self._db_connection.close()
+
+    def create_tables(self):
+        """ If no sqlite database exists with the correct tables we can create
+        one using this method.
+
+        Returns
+        -------
+        bool
+            Successfully created a new database with specified tables.
+        """
+
+        obj_schema = make_minimal_dia_object_schema()
+        src_schema = make_minimal_dia_source_schema()
+
+        self._db_cursor.execute(
+            'select name from sqlite_master where type = "table"')
+        db_tables = self._db_cursor.fetchall()
+
+        if db_tables:
+            return False
```

Comment:
I've tested this in ipython again and confirmed that this is indeed an empty list.

Right now there is no way to assure that the database created for this class is only editable by this class. This will likely be solved once the database hooks are done through the butler, but as KT said how this is going to happen is currently in flux.